subsys/bluetooth/mesh/cfg_srv.c

```diff
@@ -75,27 +79,69 @@ static int dev_comp_data_get(struct bt_mesh_model *model,
 	bt_mesh_model_msg_init(&sdu, OP_DEV_COMP_DATA_STATUS);
 
 	net_buf_simple_add_u8(&sdu, page);
-	if (IS_ENABLED(CONFIG_BT_MESH_COMP_PAGE_1) && page == 1) {
-		err = bt_mesh_comp_data_get_page_1(&sdu);
+
+	switch (page) {
```

Comment:
The previous `if` looked a bit more compact. Or it could be replaced with something like this:
```
static int comp_data_get_page(struct net_buf_simple *buf, size_t page, size_t offset)
{
	if (atomic_test_bit(bt_mesh.flags, BT_MESH_COMP_DIRTY) && (page == 0U || page == 1U)) {
		buf->size -= BT_MESH_MIC_SHORT;
		err = bt_mesh_comp_read(&sdu, page);
		if (err) {
			LOG_ERR("Unable to get stored composition data");
			return err;
		}

		buf->size += BT_MESH_MIC_SHORT;
		return 0;
	}

	if (page == 0 || page == 128) {
		bt_mesh_comp_data_get_page_0(buf, offset);
	} else if (page == 1 || page == 129) {
		bt_mesh_comp_data_get_page_1(buf);
	} else {
		return -EINVAL;
	}
}

...

err = comp_data_get_page(&sdu, page, 0);
if (err) {
	...
	return err;
}
```

Reply:
Yes, this could work. My main issue when starting implementing this part was that it was a bit cluttery and a lot of redundant checks 👍